src/uct/cuda/cuda_ipc/cuda_ipc_cache.c

```diff
@@ -48,6 +48,40 @@ typedef struct uct_cuda_ipc_remote_cache {
 
 uct_cuda_ipc_remote_cache_t uct_cuda_ipc_remote_cache;
 
+#if HAVE_CUDA_FABRIC
+static UCS_F_ALWAYS_INLINE int
+uct_cuda_ipc_rem_mpool_cache_hash_equal(const char *str1, const char *str2)
+{
+    return !memcmp(str1, str2, sizeof(CUmemFabricHandle));
+}
+
+/* djb2 hash function */
+static UCS_F_ALWAYS_INLINE khint_t
+uct_cuda_ipc_rem_mpool_cache_hash_func(const char *s)
+{
+    khint_t hash = 5381;
+    int c, i;
+
+    for (i = 0; i < sizeof(CUmemFabricHandle); i++) {
+        c = *s++;
+        hash = ((hash << 5) + hash) + c;
+    }
+
+    return hash;
+}
+
+KHASH_INIT(cuda_ipc_rem_mpool_cache, kh_cstr_t, CUmemoryPool*, 1,
```

Comment:
Given that string key requires alloc/free operations - wouldn't be better just to have `CUmemFabricHandle` as a key? And then treat it as a byte array in hash function

Reply:
@iyastreb This was previously the case but it ran into cache lookup issues when the actual key was freed. This is why I had to duplicate a copy of the key once per key, value insertion.

Reply:
Ok, I see. Still dynamic string does not look like the best choice, because of malloc/free overhead and associated error handling. The size is always the same for all entries, so maybe we just declare a key like that:
```C
typedef struct {
    uint8_t bytes[sizeof(CUmemFabricHandle)];
} cuda_ipc_rem_mpool_cache_key_t;
```